android/src/main/kotlin/design/codeux/biometric_storage/BiometricStoragePlugin.kt

```diff
@@ -254,13 +254,20 @@ class BiometricStoragePlugin : FlutterPlugin, ActivityAware, MethodCallHandler {
 //                ui(onError) { onError(AuthenticationErrorInfo(AuthenticationError.Failed, "biometric is valid but not recognized")) }
             }
         })
-        prompt.authenticate(BiometricPrompt.PromptInfo.Builder()
-            .setTitle(promptInfo.title)
-            .setSubtitle(promptInfo.subtitle)
-            .setDescription(promptInfo.description)
-            .setNegativeButtonText(promptInfo.negativeButton)
-            .setConfirmationRequired(promptInfo.confirmationRequired)
-            .build())
+
+        val promptBuilder = BiometricPrompt.PromptInfo.Builder()
+                .setTitle(promptInfo.title)
+                .setSubtitle(promptInfo.subtitle)
+                .setDescription(promptInfo.description)
+                .setConfirmationRequired(promptInfo.confirmationRequired)
+
+        if (!options.biometryOnly) {
+            promptBuilder.setDeviceCredentialAllowed(true)
+        } else {
+            promptBuilder.setNegativeButtonText(promptInfo.negativeButton)
+        }
```

Comment:
I think `setDeviceCredentialAllowed` is deprecated, and maybe we should make the intent more explicit:

```kotlin
if (options.biometryOnly) {
  promptBuilder
    .setAllowedAuthenticators(BIOMETRIC_WEAK | BIOMETRIC_STRONG)
    .setNegativeButtonText(promptInfo.negativeButton)
} else {
  promptBuilder
    .setAllowedAuthenticators(DEVICE_CREDENTIAL | BIOMETRIC_WEAK | BIOMETRIC_STRONG)
}
```

this way it's obvious that the flag simply adds `DEVICE_CREDENTIAL`